dotnet/keyspaces-lambda-csharp/README.md

```diff
@@ -0,0 +1,124 @@
+
+## Using Amazon Keyspaces with Lambda (C#/.NET)
+
+This is a sample project showing best practices for connecting to Amazon Keyspaces via AWS Lambda (C#/.NET).
+
+The project source includes function code and supporting resources:
+
+-   ` src/keyspaces-lambda-csharp`  - A C# .NET Core function.
+-   `template.yml`  - An AWS CloudFormation template that creates an application.
+-   `1-create-bucket.sh`,  `2-deploy.sh`, etc. - Shell scripts that use the AWS CLI to deploy and manage the application.
+
+Use the following instructions to deploy the sample application. For more information on the application's architecture and implementation, see  [Managing Spot Instance Requests](https://docs.aws.amazon.com/lambda/latest/dg/services-ec2-tutorial.html)  in the developer guide.
+
+## Requirements
+-   [.NET Core SDK 3.1](https://dotnet.microsoft.com/download/dotnet-core/3.1)
+-   [AWS extensions for .NET CLI](https://github.com/aws/aws-extensions-for-dotnet-cli)
+-   The Bash shell. For Linux and macOS, this is included by default. In Windows 10, you can install the  [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10)  to get a Windows-integrated version of Ubuntu and Bash.
+-   [The AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html)  v1.17 or newer.
+
+If you use the AWS CLI v2, add the following to your  [configuration file](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)  (`~/.aws/config`):
+
+```
+cli_binary_format=raw-in-base64-out
+
+```
+
+This setting enables the AWS CLI v2 to load JSON events from a file, matching the v1 behavior.
+
+## Running Tests
+
+### 0. Prerequisites
+
+Before submitting this job to Lambda it is recommended that you verify it locally on your developer machine.
+
+### 1. Environment Variables (Developer Machine)
+
+To connect to Amazon Keyspaces using Transport Layer Security (TLS), you need an Amazon digital certificate (`AmazonRootCA1.pem`) and configure the following environment variables: 
+
+- `SERVICE_USER_NAME` and `SERVICE_PASSWORD`: should match the user name and password you obtained when you generated the service-specific credentials by following the steps to  [Generate Service-Specific Credentials](https://docs.aws.amazon.com/keyspaces/latest/devguide/programmatic.credentials.html#programmatic.credentials.ssc).
+- `CERTIFICATE_PATH`: absolute path to the Amazon digital certificate file (`AmazonRootCA1.pem`).
+
+### 2. Run the tests
+
+Then execute the test by using the .NET Core CLI
+
+```sh
+$ dotnet test test/keyspaces-lambda-csharp.Tests/
+```
+## Running the Sample Application
+### 0. Prerequisites
+There are some provided scripts for you to install, deploy and configure Lambda to your existing AWS account.
+
+Before running the scripts you need to load a profile file (or set environment variables) that grant access to AWS to create S3, IAM roles, and to setup Lambda functions.   
```

Comment:
would be good to include some links here for credentials (aws profiles etc).

Reply:
Done.